core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/MetricsServiceImpl.java

```diff
@@ -227,6 +231,30 @@ void unloadDataRetentions() {
         dataRetentions.clear();
     }
 
+    private void initSystemTenant() {
+        CountDownLatch latch = new CountDownLatch(1);
+        Trigger trigger = new RepeatingTrigger.Builder().withInterval(30, MINUTES).withDelay(30, MINUTES).build();
+        dataAccess.insertTenant(new Tenant(SYSTEM_TENANT_ID))
+                .filter(ResultSet::wasApplied)
+                .map(row -> taskScheduler.scheduleTask(CreateTenants.TASK_NAME, SYSTEM_TENANT_ID, 100, emptyMap(),
```

Comment:
Correct. We only need to perform the work done by CreateTenants if a tenant does not exist. If we do not have a scheduled, background job, when would we check? When inserting data points? We could do that but reads before writes, particularly in high write volume scenarios, is an anti-pattern. CreateTenants runs every 30 minutes. The tenants_by_time keeps track of any tenants having data inserted during the past 30 minutes.